lib/spack/spack/test/concretize_preferences.py

```diff
@@ -116,9 +116,23 @@ def test_preferred_providers(self):
 
     def test_develop(self):
         """Test concretization with develop version"""
-        spec = Spec('builtin.mock.develop-test')
+        spec = Spec('develop-test')
         spec.concretize()
         assert spec.version == spack.spec.Version('0.2.15')
+        spec = Spec('develop-test2')
+        spec.concretize()
+        assert spec.version == spack.spec.Version('0.2.15')
+
+        # now add packages.yaml with develop version
+        conf = syaml.load("""\
+develop-test:
```

Comment:
@alalazo @tgamblin :  i need another advice please. When doing conretization below (see https://github.com/LLNL/spack/blob/develop/lib/spack/spack/concretize.py#L170-L179), `yaml_spec` and `yaml_index` are both empty, below is the output of
```
        print (spec.name)
        print (yaml_specs)
        print (yaml_index)
```
-->
```
develop-test
[]
{}
```
, so for some reason this `packages.yaml` is not picked up. Is there some peculiarities of `syaml.load()` that would prevent it from working here?